src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -23,14 +24,29 @@ class SignTransactionApi extends TopLevelApi {
         parsedRequest.recipientLabel = this.parseLabel(request.recipientLabel);
         parsedRequest.transaction = this.parseTransaction(request);
         parsedRequest.layout = this.parseLayout(request.layout);
-        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT) {
+        if (parsedRequest.layout === SignTransactionApi.Layouts.CHECKOUT
+            && request.layout === SignTransactionApi.Layouts.CHECKOUT) {
```

Comment:
Why do we need to check the layout for both the parsed and unparsed request?

Reply:
I have the same question.

Reply:
It's required for ts to be able to infer that the checkout specific properties exist on `request` and parsedRequest`.